src/components/DpButton/DpButtonIcon.vue

```diff
@@ -0,0 +1,27 @@
+<template>
```

Comment:
- Is this needed as a separate component?
- Shouldn't `icon` be a required prop?

Reply:
> Is this needed as a separate component?

Not really, but i felt it reduces duplication a little bit. It is not exposed as a separate component, only used in DpButton.

> Shouldn't icon be a required prop?

No, as the `v-if="icon"` (hence, the decision if the icon should be displayed) is implemented here.

Reply:
But does it makes sense to implement it inside the component? If there is no icon, the component is not needed at all, therefore I'd make the `icon` prop required.

Reply:
i just wanted to reduce redundancy of the icon and iconAfter code. implementing the v-if inside the child component enables to have the icon as one liner in the parent component like so:

```vue
<dp-button-icon :icon="icon" />
```

However if you feel that this harms readability, i'd be happy to move the code back into DpButton and remove DpButtonIcon.

Reply:
Since the component is not exposed, I don't have a strong opinion on this (so we may leave it as is). But normally I'd say it's too implicit and makes it more difficult to understand what's going on

Reply:
removed it in 8667073c15e14ea8cf77079cac60f5e11a0488fa